### PR TITLE
Security/access control

### PR DESCRIPTION
**Pull request by gafderks**
Secured API with tokens for authentication and created an ACL for authorization.

### CODE REVIEW

File: config.dist.php [hunk 1]
```diff
@@ -1,34 +1,92 @@
...
+     * Name of the view template.
+     * The template files need to be located in /view/{template}
+     *
+     * Default value: 'flatly'
+     */
     'template' => 'flatly',
+
+    /**
+     * Domain that the application will run on.
+     * Without trailing slash.
```
Comment (mhverbakel) [inline config.dist.php]: Maybe specify an example value? What if the application is not running at the root level?

File: config.dist.php [hunk 2]
```diff
@@ -1,34 +1,92 @@
...
+     */
     'domain' => 'put domain URL here',
+
+    /**
+     * Public URL of the /public folder.
+     * Without trailing slash.
```
Comment (mhverbakel) [inline config.dist.php]: Example value? Can this be a different domain?

File: src/Acl/Acl.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     * AclService constructor.
+     */
+    public function __construct() {
+        // Application roles
+        $this->addRole('guest');
+        /* member role "extends" guest, meaning the member role will get all of
+         the guest role permissions by default */
+        $this->addRole('member', 'guest');
+        $this->addRole('moderator', 'member');
+        $this->addRole('admin');
```
Comment (mhverbakel) [inline src/Acl/Acl.php]: Admin does not extend anything? So a guest can be an Admin, but not a member?

File: src/Acl/Acl.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        $this->addResource('login');
+        $this->addResource('explorer');
+        $this->addResource('activity');
+        $this->addResource('ownActivity');
+
+
+
+        // Application permissions
+        /* Now we allow or deny a role's access to resources. The third
+        argument is 'privilege'. We are using HTTP method as 'privilege'. */
```
Comment (mhverbakel) [inline src/Acl/Acl.php]: You are using what HTTP method as a privilege?

**Comment by gafderks** [comment]
@mhverbakel I have tackled the problems that you mentioned.

**Pull request merged**